kubectl-fdb/cmd/fix_coordinator_ips.go

```diff
@@ -163,44 +166,97 @@ func buildClusterFileUpdateCommands(cluster *fdbv1beta2.FoundationDBCluster, kub
 
 // updateIPsInConnectionString updates the connection string in the cluster
 // status by replacing old coordinator IPs with the latest IPs.
-func updateIPsInConnectionString(cluster *fdbv1beta2.FoundationDBCluster) error {
+func updateIPsInConnectionString(ctx context.Context, cmd *cobra.Command, cluster *fdbv1beta2.FoundationDBCluster, kubeClient client.Client) error {
 	connectionString, err := fdbv1beta2.ParseConnectionString(cluster.Status.ConnectionString)
 	if err != nil {
 		return err
 	}
+
+	// Fetch the associated process group from the coordinator address.
+	coordinatorProcessGroup := map[string]*fdbv1beta2.ProcessGroupStatus{}
+	for _, coordinator := range connectionString.Coordinators {
+		coordinatorAddress, err := fdbv1beta2.ParseProcessAddress(coordinator)
+		if err != nil {
+			return err
+		}
+
+		var processGroupFound bool
+		for _, processGroup := range cluster.Status.ProcessGroups {
+			if processGroupFound {
+				break
+			}
+			for _, address := range processGroup.Addresses {
+				if address == coordinatorAddress.IPAddress.String() {
+					coordinatorProcessGroup[coordinatorAddress.MachineAddress()] = processGroup
+
+					cmd.Println(coordinatorAddress.MachineAddress(), "is associated with process group:", processGroup.ProcessGroupID)
+					processGroupFound = true
+				}
+				break
+			}
+		}
+	}
+
+	// Update the new coordinators
 	newCoordinators := make([]string, len(connectionString.Coordinators))
 	for coordinatorIndex, coordinator := range connectionString.Coordinators {
 		coordinatorAddress, err := fdbv1beta2.ParseProcessAddress(coordinator)
 		if err != nil {
 			return err
 		}
-		for _, processGroup := range cluster.Status.ProcessGroups {
+
+		processGroup, ok := coordinatorProcessGroup[coordinatorAddress.MachineAddress()]
+		if !ok {
+			// Keep the old address if the coordinator process group is missing.
+			newCoordinators[coordinatorIndex] = coordinatorAddress.String()
+			cmd.Println("ProcessGroup for", coordinatorAddress.MachineAddress(), "is missing in the FoundationDBCluster status, coordinator address will be ignored")
```

Comment:
The message reads a bit odd to me, IIUC it's not quite ignored- it's just not updated with a correct address? Unless something in the FDB logic will cause the no-longer-existing address to get dropped?

Reply:
It's ignored for the purpose of the command, but I agree, the wording should be improved 👍